src/blueapi/plugins/data_writing.py

```diff
@@ -0,0 +1,106 @@
+import itertools
+from abc import ABC, abstractmethod, abstractproperty
+from pathlib import Path
+from typing import Optional
+
+import bluesky.plan_stubs as bps
+from aiohttp import ClientSession
+from bluesky.utils import make_decorator
+
+from blueapi.core import MsgGenerator
+from blueapi.plugins.data_writing_server import DataCollection
+
+from .data_writing_server import DataCollection, DataCollectionSetupResult
+
+DATA_COLLECTION_NUMBER = "data_collection_number"
```

Comment:
Is there a reason to favour data_collection_number over data_session, which is an existing field on StartDocument? From the event_model docs: "An optional field for grouping runs. The meaning is not mandated, but this is a data management grouping and not a scientific grouping. It is intended to group runs in a visit or set of trials."

Reply:
Nope, data session is fine, finer details of the API can be changed, this is just a rough sketch